### PR TITLE
[CH10267] Add ability to set apiKey after instance has been defined (setter)

### PR DESCRIPTION
**Pull request by sblaurock**
3 tests added, all pass.

### CODE REVIEW

**Comment by sblaurock** [comment]
> Looks good to me. Should we also add tests around browse, autosuggest and recommendations as well?

Sure, why not. Was debating doing so but thought it might be overkill. Will do!

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 1]
```diff
@@ -109,6 +109,21 @@ public ConstructorIO(String apiToken, String apiKey, boolean isHTTPS, String hos
...
     }
 
+    /**
+     * Sets apiKey
+     *
+     * @return true
+     * @throws ConstructorException
+     */
+    public boolean setApiKey(String apiKey) throws ConstructorException {
+        try {
```
Comment (rubikzube) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:119]: A try/catch here appears to be a little heavy.  I might suggest just `this.apiKey = apiKey;`

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 2]
```diff
@@ -109,6 +109,16 @@ public ConstructorIO(String apiToken, String apiKey, boolean isHTTPS, String hos
...
     }
 
+    /**
+     * Sets apiKey
+     *
+     * @return true
+     */
+    public boolean setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+        return true;
```
Comment (rubikzube) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:119]: why return true?

Comment (sblaurock) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:119]: What does Java return by default? Just remove this?

Comment (sblaurock) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:119]: > why return true?

Should this just be removed?

**Pull request merged**